src/gui/transferlistsortmodel.cpp

```diff
@@ -253,8 +239,17 @@ bool TransferListSortModel::lowerPositionThan(const QModelIndex &left, const QMo
     }
 
     // Sort according to TR_SEED_DATE
-    const QDateTime dateL = model->data(model->index(left.row(), TorrentModelItem::TR_SEED_DATE)).toDateTime();
-    const QDateTime dateR = model->data(model->index(right.row(), TorrentModelItem::TR_SEED_DATE)).toDateTime();
+    return dateLessThan(TorrentModelItem::TR_SEED_DATE, left, right);
+}
+
+// Every time we compare QDateTimes we need a fallback comparison in case both
+// values are empty. This is a workaround for unstable sort in QSortFilterProxyModel
+// (detailed discussion in #2526 and #2158).
+bool TransferListSortModel::dateLessThan(const int dateColumn, const QModelIndex &left, const QModelIndex &right) const
+{
+    const TorrentModel *model = dynamic_cast<TorrentModel*>(sourceModel());
+    const QDateTime dateL = model->data(model->index(left.row(), dateColumn)).toDateTime();
+    const QDateTime dateR = model->data(model->index(right.row(), dateColumn)).toDateTime();
     if (dateL.isValid() && dateR.isValid()) {
         if (dateL != dateR)
             return dateL < dateR;
```

Comment:
the code below this isn't equal to the one you deleted. Invalid dates should be sorted to the bottom.